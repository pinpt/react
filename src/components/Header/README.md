### Displays the title of your site, the description, and a subscribe button, and can contain search and your logo (coming soon)

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-header) | [Github](https://github.com/pinpt/react/tree/master/src/components/Header)

#### Theming

The parent className for styling this component is `.Header`

<details>
	<summary>Base Theme Styles</summary>

```css
.headerWrapper {
	background-color: var(--header-bg-color);
}

.Header {
	@apply flex py-5 flex-col;
}

.Header .top {
	@apply flex justify-between;
}

.Header .right {
	@apply flex gap-x-4;
}

.Header .center {
	@apply py-8 flex flex-col content-center text-center;
}

.Header .title {
	@apply font-semibold text-4xl md:text-5xl py-6 transition-colors duration-200;
	color: var(--header-title-color);
}

.Header .description {
	color: var(--header-description-color);
}

.Header .action {
	@apply mt-6 mx-auto;
}

.Header .Subscribe {
	color: var(--header-subscribe-color);
}
```

</details>
