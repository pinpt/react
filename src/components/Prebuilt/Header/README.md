# Prebuilt Header

The prebuilt components are the easiest to use since they work right out of the box, but provide super simple customization options. As its simplest, a prebuilt component can be rendered as such:

```jsx
import { Prebuilt, useSite } from '@pinpt/react';

const { Header } = Prebuilt;

const MySiteHeader = () => {
	const site = useSite();

	return <Header site={site} entries={entries} />;
};
```

That's it! You'll get a fully functioning header rendered with your data. Every part of the prebuilt component can be overridden, and if you want to hide a part of the app, simply return an empty fragment from the renderer:

```jsx
import { Prebuilt, useSite } from '@pinpt/react';

const { Header } = Prebuilt;

const MySiteHeader = () => {
	const site = useSite();

	return <Header site={site} renderLogo={() => <img src="path/to/my/awesome/logo" />} renderThemeToggler={() => <></>} />;
};
```
