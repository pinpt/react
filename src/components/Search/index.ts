import Results from './Results';
import Bar from './Bar';

export default {
	Results,
	Bar,
};
