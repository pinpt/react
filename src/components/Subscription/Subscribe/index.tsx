import Form from '../../Form/form';
import Field from '../../Form/field';
import React, { ChangeEvent, useCallback } from 'react';
import useEmailValidation from '../../../lib/hooks/useEmailValidation';

export interface ISubscribeProps {
	className?: string;
	name?: string;
}

const validate = async () => true;

const Subscribe = (props: ISubscribeProps) => {
	const { className = '', name } = props;
	const { email, setEmail, statusText, emailValid } = useEmailValidation(validate);

	const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
		setEmail(e.target.value);
	}, []);

	return (
		<div className={`Pinpoint SubscriptionSubscribe Wrapper ${className}`}>
			<Form title={`Subscribe to Updates${name ? ` for ${name}` : ''}.`} description="" buttonLocation="bottom" buttons={(
				<button disabled={!emailValid} className="Pinpoint SubscriptionSubscribe Submit">Submit</button>
			)}>
				<Field label="Email">
					<>
						<input
							value={email}
							className="Pinpoint SubscriptionSubscribe Field"
							placeholder="Enter your email address"
							onChange={handleChange}
						/>
						{statusText}
					</>
				</Field>
			</Form>
		</div>
	);
};

export default Subscribe;