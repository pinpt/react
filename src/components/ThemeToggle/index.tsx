import React, { useCallback } from 'react';

export interface IThemeToggleProps {
	className?: string;
	onThemeChange?: (theme: 'light' | 'dark') => void;
	localStorageKey?: string; // if you want to change the localstorage key. defaults to 'theme'
}

const ThemeToggle = (props: IThemeToggleProps) => {
	const { className = '', onThemeChange, localStorageKey = 'theme' } = props;

	const handleToggleTheme = useCallback(() => {
		const element = document.getElementsByTagName('html')?.[0];
		if (element) {
			if (element.classList.contains('dark')) {
				element.classList.remove('dark');
				onThemeChange?.('light');
				window.localStorage.setItem(localStorageKey, 'light');
			} else {
				element.classList.add('dark');
				onThemeChange?.('dark');
				window.localStorage.setItem(localStorageKey, 'dark');
			}
		}
	}, [onThemeChange, localStorageKey]);

	React.useEffect(() => {
		if (typeof window !== 'undefined') {
			const localTheme = window.localStorage.getItem(localStorageKey);
			if (localTheme) {
				const element = document.getElementsByTagName('html')?.[0];
				switch (localTheme) {
					case 'dark': {
						element.classList.add('dark');
						onThemeChange?.('dark');
						break;
					}
					case 'light': {
						element.classList.remove('dark');
						onThemeChange?.('light');
						break;
					}
					default:
						break;
				}
			}
		}
	}, [localStorageKey, onThemeChange]);

	return (
		<div className={`Pinpoint ThemeToggle ${className}`}>
			<button className="mode" onClick={handleToggleTheme} aria-label="Toggle Theme" />
		</div>
	);
};

export default ThemeToggle;
