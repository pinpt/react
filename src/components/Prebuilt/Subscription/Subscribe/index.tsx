import React, { ReactElement } from 'react';
import { ISite } from '../../../../lib/types/site';
import { ICopyrightProps } from '../../../Copyright';
import { IFooterProps } from '../../../Footer';
import { IHeaderProps } from '../../../Header';
import withWrapper from '../../../Internal/withWrapper';
import { ILogoProps } from '../../../Logo';
import { ISocialMediaBarProps } from '../../../SocialMedia/SocialMediaBar';
import SubscribeComponent from '../../../Subscription/Subscribe';
import { IThemeToggleProps } from '../../../ThemeToggle';
import Footer from '../../Footer';
import Header from '../../Header';

export interface IPrebuiltSubscribeProps {
	className?: string;
	site: ISite;
	renderHeader?: (site: ISite) => ReactElement<IHeaderProps>;
	handleSelectHome?: () => void;
	renderLogo?: (site: ISite) => ReactElement<ILogoProps>;
	renderThemeToggle?: (site: ISite) => ReactElement<IThemeToggleProps>;
	renderFooter?: (site: ISite) => ReactElement<IFooterProps>;
	renderSocial?: (site: ISite) => ReactElement<ISocialMediaBarProps>;
	renderCopyright?: (site: ISite) => ReactElement<ICopyrightProps>;
	handleSubmit?: (email: string) => Promise<any>;
	contentClassName?: string;
	pending?: boolean;
}

const Subscribe = (props: IPrebuiltSubscribeProps) => {
	const {
		className = '',
		renderHeader,
		site,
		handleSelectHome,
		renderLogo,
		renderThemeToggle,
		renderFooter,
		renderCopyright,
		renderSocial,
		handleSubmit,
		contentClassName,
		pending,
	} = props;

	return (
		<div className={`Pinpoint Page SubscriptionSubscribe Prebuilt ${className}`}>
			{withWrapper(
				renderHeader?.(site) ?? (
					<Header
						site={site}
						handleSelectHome={handleSelectHome}
						renderLogo={renderLogo}
						renderSearch={() => <></>}
						renderSubscribe={() => <></>}
						renderThemeToggle={renderThemeToggle}
					/>
				),
				'header'
			)}
			{withWrapper(
				<SubscribeComponent
					name={site.name}
					handleSubmit={handleSubmit}
					className={`Prebuilt ${contentClassName}`}
					pending={pending}
				/>,
				'subscribe'
			)}
			{withWrapper(
				renderFooter?.(site) ?? (
					<Footer
						site={site}
						renderCopyright={renderCopyright}
						renderLogo={renderLogo}
						renderSocial={renderSocial}
						renderSubscribe={() => <></>}
					/>
				),
				'footer'
			)}
		</div>
	);
};

export default Subscribe;
