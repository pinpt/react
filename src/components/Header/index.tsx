import { ReactElement } from 'react';
import { ISearchBarProps } from '../Search/Bar';
import { ISubscribeProps } from '../Subscribe';
import { IThemeToggleProps } from '../ThemeToggle';

export interface IHeaderProps {
	className?: string;
	title?: string;
	description?: string;
	subscribe?: ReactElement<ISubscribeProps>;
	themeToggle?: ReactElement<IThemeToggleProps>;
	search?: ReactElement<ISearchBarProps>;
}

const Header = (props: IHeaderProps) => {
	const { className = '', title, description, subscribe, themeToggle, search } = props;

	return (
		<div className={`Pinpoint Header ${className}`}>
			<div className="top">
				<div className="left"></div>
				<div className="right">
					{search}
					{themeToggle}
				</div>
			</div>
			<div className="center">
				{title && <h1 className="title">{title}</h1>}
				{description && <div className="description">{description}</div>}
				{subscribe && <div className="action">{subscribe}</div>}
			</div>
		</div>
	);
};

export default Header;
