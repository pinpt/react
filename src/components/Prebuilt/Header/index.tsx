import { ReactElement } from 'react';
import { ISite } from '../../../lib';
import BaseHeader from '../../Header';
import Logo, { ILogoProps } from '../../Logo';
import Search from '../../Search';
import { ISearchBarProps } from '../../Search/Bar';
import Subscribe, { ISubscribeProps } from '../../Subscribe';
import ThemeToggle, { IThemeToggleProps } from '../../ThemeToggle';

export interface IPrebuiltHeaderProps {
	className?: string;
	searchTerm?: string;
	site: ISite;
	handleSearch?: (value: string) => void;
	handleSelectHome?: () => void;
	renderLogo?: (site: ISite) => ReactElement<ILogoProps>;
	renderSearch?: (site: ISite) => ReactElement<ISearchBarProps>;
	renderSubscribe?: (site: ISite) => ReactElement<ISubscribeProps>;
	renderThemeToggle?: (site: ISite) => ReactElement<IThemeToggleProps>;
	title?: string;
	description?: string;
	onToggleMenu?: () => void;
	mobileMenu?: boolean;
}

const Header = (props: IPrebuiltHeaderProps) => {
	const {
		className = '',
		searchTerm,
		site,
		handleSearch,
		handleSelectHome,
		renderLogo,
		renderSearch,
		renderSubscribe,
		renderThemeToggle,
		title,
		description,
		onToggleMenu,
		mobileMenu,
	} = props;

	return (
		<BaseHeader
			className={`Prebuilt ${className}`}
			title={title ?? site.theme?.title ?? site.name}
			description={description ?? site.theme?.description ?? site.name}
			onClick={() => handleSelectHome?.()}
			subscribe={renderSubscribe?.(site) ?? <Subscribe className="Prebuilt" href="/subscription/subscribe" />}
			themeToggle={renderThemeToggle?.(site) ?? <ThemeToggle className="Prebuilt" />}
			search={
				renderSearch?.(site) ?? (
					<Search.Bar defaultValue={searchTerm} onSubmit={handleSearch} className="Prebuilt" />
				)
			}
			onToggleMenu={onToggleMenu}
			mobileMenu={mobileMenu}
			logo={
				renderLogo?.(site) ?? (
					<Logo
						src={site.logoUrl}
						className="Prebuilt"
						href={site.theme?.logoLink ?? site.url}
						title="Header Logo Home Page"
					/>
				)
			}
		/>
	);
};

export default Header;
