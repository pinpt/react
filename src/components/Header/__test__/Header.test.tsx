import renderer from 'react-test-renderer';
import Header from '..';
import Search from '../../Search';
import Subscribe from '../../Subscribe';
import ThemeToggle from '../../ThemeToggle';

test('Test complete', () => {
	const component = renderer.create(
		<Header
			subscribe={<Subscribe href="https://pinpoint.com" />}
			title="Pinpoint Demo Changelog"
			description="See what's new in the Pinpoint Demo."
			themeToggle={<ThemeToggle />}
			search={<Search.Bar />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test empty', () => {
	const component = renderer.create(<Header />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no search', () => {
	const component = renderer.create(
		<Header
			subscribe={<Subscribe href="https://pinpoint.com" />}
			title="Pinpoint Demo Changelog"
			description="See what's new in the Pinpoint Demo."
			themeToggle={<ThemeToggle />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no theme', () => {
	const component = renderer.create(
		<Header
			subscribe={<Subscribe href="https://pinpoint.com" />}
			title="Pinpoint Demo Changelog"
			description="See what's new in the Pinpoint Demo."
			search={<Search.Bar />}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no description', () => {
	const component = renderer.create(
		<Header subscribe={<Subscribe href="https://pinpoint.com" />} title="Pinpoint Demo Changelog" />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no subscribe', () => {
	const component = renderer.create(
		<Header title="Pinpoint Demo Changelog" description="See what's new in the Pinpoint Demo." />
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test title only', () => {
	const component = renderer.create(<Header title="Pinpoint Demo Changelog" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
