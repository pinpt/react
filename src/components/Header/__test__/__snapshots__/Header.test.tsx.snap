// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test complete 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    >
      <div
        className="Pinpoint Search Bar "
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          height={16}
          role="img"
          viewBox="0 0 512 512"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
        </form>
      </div>
      <div
        className="Pinpoint ThemeToggle "
      >
        <button
          className="mode"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
    <div
      className="description"
    >
      See what's new in the Pinpoint Demo.
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Subscribe "
        href="https://pinpoint.com"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test empty 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    />
  </div>
  <div
    className="center"
  />
</div>
`;

exports[`Test no description 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    />
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Subscribe "
        href="https://pinpoint.com"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test no search 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    >
      <div
        className="Pinpoint ThemeToggle "
      >
        <button
          className="mode"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
    <div
      className="description"
    >
      See what's new in the Pinpoint Demo.
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Subscribe "
        href="https://pinpoint.com"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test no subscribe 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    />
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
    <div
      className="description"
    >
      See what's new in the Pinpoint Demo.
    </div>
  </div>
</div>
`;

exports[`Test no theme 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    >
      <div
        className="Pinpoint Search Bar "
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          height={16}
          role="img"
          viewBox="0 0 512 512"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search"
            value=""
          />
        </form>
      </div>
    </div>
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
    <div
      className="description"
    >
      See what's new in the Pinpoint Demo.
    </div>
    <div
      className="action"
    >
      <a
        className="Pinpoint Internal Link Subscribe "
        href="https://pinpoint.com"
      >
        <svg
          aria-hidden="true"
          className="icon"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          height={14}
          role="img"
          viewBox="0 0 512 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
          />
        </svg>
        <span
          className="text"
        >
          Subscribe to Updates
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test title only 1`] = `
<div
  className="Pinpoint Header "
>
  <div
    className="top"
  >
    <div
      className="left"
    />
    <div
      className="right"
    />
  </div>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Pinpoint Demo Changelog
    </h1>
  </div>
</div>
`;
